include/swift/Frontend/Frontend.h

```diff
@@ -213,8 +214,11 @@ class CompilerInvocation {
   /// Computes the runtime resource path relative to the given Swift
   /// executable.
   static void computeRuntimeResourcePathFromExecutablePath(
-      StringRef mainExecutablePath,
-      llvm::SmallString<128> &runtimeResourcePath);
+      StringRef mainExecutablePath, bool shared,
+      llvm::SmallVectorImpl<char> &runtimeResourcePath);
+
+  /// Appends `lib/swift[_static]` to the given path
+  static void appendLibDir(llvm::SmallVectorImpl<char> &path, bool shared);
```

Comment:
Would be nice to make this `appendSwiftLibDir` since `libdir` in most contexts here tends to refer to `lib` vs `lib32` or `lib64`.